feat: add ClusterOriginIssuer

terinjokes opened the pull request:
When using this controller in single-tenant clusters it becomes desirable for many users to have a single cluster-scoped issuer rather than each namespace duplicating the issuers and secrets.

This changeset introduces a "cluster resource namespace", which defaults to the same namespace as the controller (eg, "origin-ca-issuer"). A ClusterOriginIssuer resource type has been added which resolves SecretKeySelectors in this namespace. Other than determing what namespace to resolve the secret, the implementation is the same as the existing OriginIssuer.

Fixes: #24

cbroglie commented on pkgs/controllers/certificaterequest.go:
```diff
@@ -110,32 +111,68 @@ func (r *CertificateRequestController) Reconcile(ctx context.Context, cr *certma
-	if err := r.Client.Get(ctx, issNamespaceName, &iss); err != nil {
-		log.Error(err, "failed to retrieve OriginIssuer resource", "namespace", issNamespaceName.Namespace, "name", issNamespaceName.Name)
-		_ = r.setStatus(ctx, cr, cmmeta.ConditionFalse, certmanager.CertificateRequestReasonPending, fmt.Sprintf("Failed to retrieve OriginIssuer resource %s: %v", issNamespaceName, err))
+	switch cr.Spec.IssuerRef.Kind {
```
Does it make sense to include a default case which returns an error for an unexpected kind?

terinjokes committed:
feat: add ClusterOriginIssuer

When using this controller in single-tenant clusters it becomes
desirable for many users to have a single cluster-scoped issuer rather
than each namespace duplicating the issuers and secrets.

This changeset introduces a "cluster resource namespace", which defaults
to the same namespace as the controller (eg, "origin-ca-issuer"). A
ClusterOriginIssuer resource type has been added which resolves
SecretKeySelectors in this namespace. Other than determing what
namespace to resolve the secret, the implementation is the same as the
existing OriginIssuer.

Fixes: #24
